src/app/components/layout/side-menu/side-menu.component.ts

```diff
@@ -117,10 +117,17 @@ export class SideMenuComponent implements OnInit {
     // place the submenu in the correct position relevant to the menu item
     submenuWrapper.style.top = (menuItemPosition.top - addedMargin).toString() + 'px';
     submenuWrapper.style.left = (menuItemPosition.width + 4).toString() + 'px';
+    // delay the submenu open by 550ms to avoid flickering
     setTimeout(function () {
+      // If mouse is not over the menu or child sub-menu then return and don't open the sub-menu   
+      if (!sidemenuLink.matches(':hover')) {
+        return;
+      }
+      if (!submenuWrapper.classList.contains('show')) {
+        this.toggleSubmenuState(menuLink);
+      }
       submenuWrapper.classList.add('show');
-      this.toggleSubmenuState(menuLink);
-    }.bind(this), 200);
+    }.bind(this), 500);
```

Comment:
actually you can next to it as

bind(this), 500);  // Added a delay to prevent the submenu from opening too quickly when hovering over the parent menu item, and ensured it only opens when focused for some milliseconds.